getDataTableReader.html

```diff
@@ -0,0 +1,154 @@
+      }
+
+      // Get the underlying data using the DataTableReader
+      async function getDataAsync (pageNumber) {
+        // Get data from a specified sheet
+        const sheet = viz.workbook.activeSheet.worksheets.find(
+          sheet => sheet.name === 'Storm Map Sheet'
+        );
+        const datasource = await sheet.getDataSourcesAsync();
+
+        // Get the logical tables from the first data source
+        const logicalTables = await datasource[0].getLogicalTablesAsync();
+
+        // Loop through each table that was used in creating this data source
+        logicalTables.forEach(function (table) {
+          console.log(`Name of logical table: ${table.caption}`);
+          document.getElementById(
+            'info'
+          ).innerHTML = `<p>Name of logical table: ${table.caption}</p>`
+        });
+
+        // Get the data from the first logical table
+        const logicalTableData = await datasource[0].getLogicalTableDataAsync(
+          logicalTables[0].id
+        );
+        console.log(logicalTableData);
+        console.log(`Total row count: ${logicalTableData.totalRowCount}`);
+        // Display the row count
+        document
+          .getElementById('info')
+          .insertAdjacentHTML(
+            'beforeend',
+            `<p>Total row count: ${logicalTableData.totalRowCount}</p>`
+          );
+
+        // Set the page size to 1000 for this sample; the default value is 10000 rows
+        const pageRowCount = 1000;
+        // Create the data table reader
+        const pageReader = await datasource[0].getLogicalTableDataReaderAsync(
+          logicalTables[0].id,
+          pageRowCount
+        );
+        // Display the number of pages to read
+        console.log(`Number of pages: ${pageReader.pageCount}`);
+        document
+          .getElementById('info')
+          .insertAdjacentHTML(
+            'beforeend',
+            `<p>Number of pages: ${pageReader.pageCount}</p>`
+          );
+
+        // Get the page data based on page number requested
+        let currentPageDataTable = await pageReader.getPageAsync(pageNumber);
+        console.log(currentPageDataTable);
+
+
+        // Add data to HTML element.
+        const target = document.getElementById('dataTarget')
+        target.innerHTML = `<h4>Data Table Page ${pageNumber} Data:</h4><p>${JSON.stringify(
+          currentPageDataTable.data
+        )}</p>`;
+
+        // Free up memory when done
+        await pageReader.releaseAsync();
+      }
+
+      // get the viz from the HTML web component
+      const viz = document.getElementById('tableauViz');
+
+      // Event fired when a viz first becomes interactive.
+      viz.addEventListener(
+        TableauEventType.FirstInteractive,
+        handleFirstInteractive
+      );
+    </script>
+  </head>
+
+  <body>
+    <div>
+      <h1>Get Logical Data (TableDataReader) Example </h1>
+      <p>
+        Click <b>Get Data</b> to get the first page (page 0) of the logical
```

Comment:
Shouldn't this be: get the selected page of the logical... since the user can pick the page to select.

Reply:
Yes. That's how it should work. However, I couldn't get it work right earlier. I can try again.

Reply:
@anyoung-tableau  I was having trouble figuring out how to read the page selected from the page selector drop down list and use it as input with the **Get Data** button click.  
As it is now, all I could get to work is to have the **Get Data** button gets the first page (page 0) of data. And after that anytime a user selects the a page from the page selector, the page is retrieved.

Reply:
@johnDance with @anyoung-tableau help I was able to get this to work correctly (logically?).